test/common/buffer/buffer_test.cc

```diff
@@ -0,0 +1,652 @@
+#include <limits>
+
+#include "envoy/common/exception.h"
+
+#include "common/buffer/buffer_impl.h"
+
+#include "test/common/buffer/utility.h"
+#include "test/test_common/printers.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Buffer {
+namespace {
+
+TEST(BufferHelperTest, PeekI8) {
```

Comment:
Along with the comment in buffer.h, if `buffer.peekLEInt<uint32_t, 2>` is to be allowed, please add tests to exercise it.

Reply:
adding